onedal/decomposition/pca.cpp

```diff
@@ -136,10 +136,13 @@ ONEDAL_PY_INIT_MODULE(decomposition) {
 
     using task_list = types<task::dim_reduction>;
     auto sub = m.def_submodule("decomposition");
-
-    ONEDAL_PY_INSTANTIATE(init_train_ops, sub, policy_list, task_list);
-    ONEDAL_PY_INSTANTIATE(init_infer_ops, sub, policy_list, task_list);
-
+    #ifdef ONEDAL_DATA_PARALLEL_SPMD
+        ONEDAL_PY_INSTANTIATE(init_train_ops, sub, policy_list_spmd, task_list);
+        ONEDAL_PY_INSTANTIATE(init_infer_ops, sub, policy_list, task_list);
```

Comment:
We don't support SPMD policy for infer?

Reply:
Just train - https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/algo/pca/detail/infer_ops_dpc.cpp